pkg/netpol/connlist/connlist.go

```diff
@@ -550,3 +569,78 @@ func (ca *ConnlistAnalyzer) logWarning(msg string) {
 		ca.logger.Warnf(msg)
 	}
 }
+
+// checkIfP2PConnOrExposureConn checks if the given connection is between two peers from the parsed resources, if yes returns it,
+// otherwise the connection belongs to exposure-analysis, will be added to the provided map
+func (ca *ConnlistAnalyzer) checkIfP2PConnOrExposureConn(pe *eval.PolicyEngine, allowedConnections common.Connection,
+	src, dst Peer, exposuresMap exposureMap) *connection {
+	if !ca.exposureAnalysis {
+		// if exposure analysis option is off , the connection is definitely a P2PConnection
+		return createConnectionObject(allowedConnections, src, dst)
+	}
+	// else exposure analysis is on
+	// TODO : enhance this if condition after implementing eval.RepresentativePeer
+	if src.Name() != common.PodInRepNs && dst.Name() != common.PodInRepNs {
+		// both src and dst are peers are found in the parsed resources
+		return createConnectionObject(allowedConnections, src, dst)
+	}
+	// else: one of the peers is inferred from a netpol-rule , and the other is a peer from the parsed resources
+	// an exposure analysis connection
+	if src.Name() != common.PodInRepNs {
+		// dst is the inferred from netpol peer, we have an exposed egress for the src peer
+		addConnToExposureMap(pe, allowedConnections, src, dst, false, exposuresMap)
+	} else {
+		// src is the inferred from netpol peer, we have an exposed ingress to the dst peer
+		addConnToExposureMap(pe, allowedConnections, src, dst, true, exposuresMap)
+	}
+	return nil
+}
+
+// helper function - returns a connection object from the given fields
+func createConnectionObject(allowedConnections common.Connection, src, dst Peer) *connection {
+	return &connection{
+		src:               src,
+		dst:               dst,
+		allConnections:    allowedConnections.AllConnections(),
+		protocolsAndPorts: allowedConnections.ProtocolsAndPortsMap(),
+	}
+}
+
+// helper function - adds a connection and its data to the exposure-analysis
+func addConnToExposureMap(pe *eval.PolicyEngine, allowedConnections common.Connection, src, dst Peer, isIngress bool,
```

Comment:
should this be a method of the type `exposureMap` instead?

Reply:
done [commit](https://github.com/np-guard/netpol-analyzer/pull/296/commits/f979481af0e9b6b3be4c5a8801f08c086623af9f)